roles/ceph-rgw/tasks/rgw_pool_pgs.yml

```diff
@@ -2,7 +2,23 @@
 - name: create rgw pools if create_pools is defined
   command: ceph --connect-timeout 5 --cluster {{ cluster }} osd pool create {{ item.key }} {{ item.value.pg_num }}
   changed_when: false
-  failed_when: false
+  run_once: true
   with_dict: "{{ create_pools }}"
   when:
-    - create_pools is defined
+    - not containerized_deployment
+
+# On first monitor.
+- name: set_fact docker_exec_rgw_cmd
+  set_fact:
+    docker_exec_rgw_cmd: "docker exec ceph-mon-{{ hostvars[groups.get(mon_group_name)[0]]['ansible_hostname'] }}"
+  when:
+    - containerized_deployment
+
+- name: create rgw pools if create_pools is defined
+  command: "{{ docker_exec_rgw_cmd }} ceph --connect-timeout 5 --cluster {{ cluster }} osd pool create {{ item.key }} {{ item.value.pg_num }}"
+  changed_when: false
+  run_once: true
+  with_dict: "{{ create_pools }}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  when:
```

Comment:
@jtudelag remove this when

Reply:
If I do this, this task will be executed always. And when non-containerized, the fact docker_exec_rgw_cmd would be undefined.

Reply:
yes, this is the idea, when non-containerized the fact `docker_exec_rgw_cmd ` won't be defined and the command won't include the prefix `docker exec ceph-mon ...` so it can be executed in non-containerized deployment.
But somehow you are right, so I guess you have to add `| default('')` like this :
`command: "{{ docker_exec_rgw_cmd }} | default('') ceph --connect-timeout 5 --cluster {{ cluster }} osd pool create {{ item.key }} {{ item.value.pg_num }}"`
nice catch, I missed this.

That way you make that task common to both containerized and non-containerized deployment and avoid kind of duplicated code so you can remove the first task.

Reply:
I can see. Isn't it a little hacky? Also, you would need to be carefull with trailing  spaces 

In that case, wouldn't it be better, at least less hacky, more readable and easy to maintain, to have a fact to set. For example `"ceph_command" : "ceph" `when non containerized, and `"ceph_comand": "docker exec..."`.  Then in the main tasks `command: "{{ ceph_command }} --cluster ...."`.

Reply:
@jtudelag At first sight, I don't see why having more task than necessary is more readable and easy to maintain, the more of code you have harder it is to maintain.

I think it's more logic to keep the common part of the command outside of the fact.
The only difference is the prefix `docker exec <name>` so this is well the variable.
By the way, it can be a bit annoying for debugging purpose to have the command split.

I'm not trying to block your suggestion, just sharing my own opinion.
So what I would propose is to keep consistency for this PR with what we already have in the playbook.
Then, if we should change the style of how we manage ceph command for containerized deployment vs. non-containerized, I think it should be addressed in a dedicated PR so it could be applied in the whole playbook, again, for consistency reasons, because I don't really see the point of having mixed styles to achieve a same goal at many places.

@jtudelag What do you think?

Reply:
@guits @leseb sorry guys, long holidays completely isolated from anything with internet ;).

**"So what I would propose is to keep consistency for this PR with what we already have in the playbook."**

Agree with that, I will go back to my first approach. I will keep Docker related tasks inside docker/ folder.

**"Then, if we should change the style of how we manage ceph command for containerized deployment vs. non-containerized, I think it should be addressed in a dedicated PR so it could be applied in the whole playbook, again, for consistency reasons, because I don't really see the point of having mixed styles to achieve a same goal at many places."**

Agree, happy to help here and propose a few ideas. Every decision has Its trade-offs, so in the end, It is a matter of being consistent with what It's been agreed ;).